### PR TITLE
feat: add exception handle in CreateOrUpdateBusinessPartnerHandler

### PR DESCRIPTION
**Pull request by msimionati**
Add exception handle in CreateOrUpdateBusinessPartnerHandler to handle the exception "User can't have more than 10 accounts in Sap" for show the correct message in log and console.

### CODE REVIEW

File: Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs [hunk 1]
```diff
@@ -29,14 +29,26 @@ public class CreateOrUpdateBusinessPartnerHandler
...
+                return !dequeuedTask.ExistentBusinessPartner.CreateDate.HasValue ?
+                    await CreateBusinessPartner(dequeuedTask, sapSystem) :
+                    await UpdateBusinessPartner(dequeuedTask, sapSystem);
+            }
+            catch (Exception ex)
+            {
+                return new SapTaskResult
+                {
+                    IsSuccessful = false,
+                    SapResponseContent = ex.Message,
```
Comment (jhoffman-ms) [inline Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs:48]: What do you think to add the user that have the error?. For example: 
```$"Failed to create or update the Business Partner for the user: {dequeuedTask.BusinessPartner.CardCode}. Error: {ex.Message}"```

File: Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs [hunk 2]
```diff
@@ -29,14 +29,26 @@ public class CreateOrUpdateBusinessPartnerHandler
...
-            return !dequeuedTask.ExistentBusinessPartner.CreateDate.HasValue ?
-                await CreateBusinessPartner(dequeuedTask, sapSystem) :
-                await UpdateBusinessPartner(dequeuedTask, sapSystem);
+                return !dequeuedTask.ExistentBusinessPartner.CreateDate.HasValue ?
+                    await CreateBusinessPartner(dequeuedTask, sapSystem) :
+                    await UpdateBusinessPartner(dequeuedTask, sapSystem);
+            }
+            catch (Exception ex)
+            {
+                return new SapTaskResult
```
Comment (nreal) [inline Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs:45]: What do you think to add a logger in standard output? for example
`_logger.LogError(e, $"Failed to create or update the Business Partner for the user: {dequeuedTask.BusinessPartner.CardCode}. Error: {e.Message}"`

Comment (msimionati) [reply Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs:52]: Implement the logger as you suggested, I had to modify several tests to make it compile and the tests run successfully

File: Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs (hunk 1 above)
Comment (msimionati) [reply Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs]: Implement the message as you suggested, I had to modify a test to set de BusinessPartner.CardCode in the mock

File: Doppler.Sap.Test/CreateOrUpdateBusinessPartnerHandlerTest.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ public class CreateOrUpdateBusinessPartnerHandlerTest
     {
         [Fact]
 
-        public void CreateOrUpdateBusinessPartnerHandler_ShouldBeThownsAnExcpetion_WhenCountryCodeNotAROrUS()
+        public async void CreateOrUpdateBusinessPartnerHandler_ShouldBeThownsAnExcpetion_WhenCountryCodeNotAROrUS()
```
Comment (nreal) [inline Doppler.Sap.Test/CreateOrUpdateBusinessPartnerHandlerTest.cs:23]: I think that if this method is `async` should return a `Task`, 
https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#avoid-async-void

Comment (msimionati) [reply Doppler.Sap.Test/CreateOrUpdateBusinessPartnerHandlerTest.cs:23]: You are right I will change it

Comment (msimionati) [reply Doppler.Sap.Test/CreateOrUpdateBusinessPartnerHandlerTest.cs]: Done

**Comment by jhoffman-ms** [comment]
👍

**Comment by nreal** [comment]
👍

**Pull request merged**